react/setup-modal.jsx

```diff
@@ -69,9 +69,11 @@ let SetupModal = React.createClass({
     return asink(function *() {
       this.close()
       let dattcore = this.props.dattcore
-      yield dattcore.asyncSetUserName(newUserName)
       yield dattcore.asyncSetUserSetupFlag(true)
+      let userNameSynced = dattcore.asyncSetUserName(newUserName) // *intentionally* NOT yield-ing
       window.location.hash = '#/frontpage'
+      yield userNameSynced
+      dattcore.emit('refresh-content', null)
```

Comment:
Only dattcore should be emitting events on the dattcore object. If the UI needs its own events, then it needs a separate system for that.